discovery-provider/src/queries/queries.py

```diff
@@ -1410,3 +1410,94 @@ def get_saves(save_type):
+        if "handle" in request.args:
+            handle = request.args.get("handle").lower()
+            base_query = base_query.filter_by(handle_lc=handle)
+        if "id" in request.args:
+            id = int(request.args.get("id"))
+            base_query = base_query.filter_by(user_id=id)
+
+        users = base_query.all()
+        users = helpers.query_result_to_list(users)
+
+        user = users[0]
+        user_id = user['user_id']
+
+        # bundle peripheral info into user results
+        users = populate_user_metadata(session, [user_id], users, user_id)
+        user = users[0]
+
+        # Get saved playlists / albums ids
+        saved_query = session.query(Save).filter(
+            Save.user_id == user_id,
+            Save.is_current == True,
+            Save.is_delete == False,
+            or_(Save.save_type == SaveType.playlist,Save.save_type == SaveType.album),
+            Save.save_item_id.in_(
+                session.query(Playlist.playlist_id).filter(
+                    Playlist.is_current == True,
+                    Playlist.is_delete == False,
+                    Playlist.is_private == False
+                )
+            )
+        )
+ 
+        saved_query_results = saved_query.all()
+        save_collections = helpers.query_result_to_list(saved_query_results)
+        save_collection_ids = [item['save_item_id'] for item in save_collections]
+
+        # Get Playlist/Albums saved or owned by the user
+        playlist_query = session.query(Playlist).filter(
+                or_(
+                    and_(Playlist.is_current == True, Playlist.is_delete == False, Playlist.playlist_owner_id == user_id),
+                    and_(Playlist.is_current == True, Playlist.playlist_id.in_(save_collection_ids))
+                )
+            ).order_by(desc(Playlist.created_at))
+        playlists = playlist_query.all()
+        playlists = helpers.query_result_to_list(playlists)
+
+        playlist_owner_ids = list(set([playlist['playlist_owner_id'] for playlist in playlists]))
+
+        # Get Users for the Playlist/Albums
+        user_query = session.query(User).filter(
+                and_(User.is_current == True, User.user_id.in_(playlist_owner_ids))
+            )
+        users = user_query.all()
+        users = helpers.query_result_to_list(users)
+        user_map = {}
+
+        collections = []
+        # Map the users to the collections
+        for collection_user in users:
+             user_map[collection_user['user_id']] = collection_user
+        for playlist in playlists:
+            collection_owner = user_map[playlist['playlist_owner_id']]
+            collections.append({
+                'id': playlist['playlist_id'],
+                'name': playlist['playlist_name'],
+                'is_album': playlist['is_album'],
+                'user': { 'id': collection_owner['user_id'], 'handle': collection_owner['handle'] }
+            })
+        user['collections'] = collections
```

Comment:
we don't have this notion of collections anywhere on the backend. let's just keep it as 'playlists'

Reply:
yeah, it's just weird that the terms playlist is overloaded w/ type playlist/album, but agreed that using the term `playlists` would be more consistent

Reply:
yeah, but for consistency sake, let's not introduce collection here. the fact that is_album is a field on the playlist model just makes this already bad :)

Reply:
^agreed